asyncscope.md

```diff
@@ -607,6 +608,82 @@ class Camera {
 }
 ```
 
+## Recursively spawning work until completion
+Below are two ways you could recursively spawn work on a scope using `let_with_async_scope` or `counting_scope`. Both
+examples are perfectly valid ways to ensure that all new work spawned from previous work will complete.
+
+:::cmptable
+
+### let_with_async_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler sch, ex::counting_scope& scope, tree& t) {
+  return ex::schedule(sch) | then([sch, &]() {
+    if (t.left)
+      ex::spawn(scope, process(sch, scope, t.left.get()));
+    if (t.right)
+      ex::spawn(scope, process(sch, scope, t.right.get()));
+    do_stuff(t.data);
+  });
+}
+
+int main() {
+  ex::scheduler sch;
+  tree t = make_tree();
+  // let_with_async_scope will ensure all new work will be spawned on the
+  // scope and will not be joined until all work is finished
+  this_thread::sync_wait(ex::let_with_async_scope([&, sch](auto scope) {
+	return process(sch, scope, t);
+  }));
+}
+```
+
+### counting_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler sch, tree& t) {
+  return ex::let_value_with([]() noexcept { return ex::counting_scope{}; }, [&](ex::counting_scope& scope) {
```

Comment:
I don't really like suggesting this as an acceptable way of spawning recursive work. It looks really ugly and the exception handling is gross. I would much rather prefer people to use let_with_async_scope, and this is basically let_with_async_scope.

Do you have a suggestion on a better way to do this using counting_scope?

Reply:
I've just deleted a draft comment that said you should just let the exception bubble out and use `finally` to join the scope in order to match the behaviour of the other example but, actually, both examples need exception handling code. `spawn` is spec'd not to accept senders that complete with an error, so the `then([] { … spawn(…); … })` sender needs to be `noexcept`, which implies either crashing when `spawn()` fails or handling those exceptions.